lib/Dialects/VectorExt/Transform/VectorWarpUtils.cpp

```diff
@@ -265,6 +267,48 @@ struct WarpOpReduction : public OpRewritePattern<WarpSingleLaneOp> {
   }
 };
 
+struct SplitVectorReduction : public OpRewritePattern<vector::ReductionOp> {
+  using OpRewritePattern<vector::ReductionOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(vector::ReductionOp reductionOp,
+                                PatternRewriter &rewriter) const override {
+    // Process only ReduceOps inside of WarpSingleLaneOps.
+    if (!isa<WarpSingleLaneOp>(reductionOp->getParentOp()))
+      return failure();
+    auto vectorType = reductionOp.vector().getType().cast<VectorType>();
+    // Only rank 1 vectors supported.
+    if (vectorType.getRank() != 1)
+      return failure();
+    // Only vectors that are larger than kWarpSize can be split.
+    int64_t vecSize = vectorType.getShape()[0];
+    if (vecSize <= kWarpSize)
+      return failure();
+
+    Location loc = reductionOp.getLoc();
+    int64_t offset = 0;
+    Value accumulator;
+    while (offset < vecSize) {
+      int64_t size = std::min<int64_t>(kWarpSize,  vecSize - offset);
+      Value vecSlice = rewriter.create<vector::ExtractStridedSliceOp>(
+          loc, reductionOp.vector(), std::initializer_list<int64_t>({offset}),
+          std::initializer_list<int64_t>({size}),
+          std::initializer_list<int64_t>({1}));
+      Value warpReduction = rewriter.create<vector::ReductionOp>(
+          loc, reductionOp.kind(), vecSlice);
+      offset += size;
+
+      if (!accumulator)
+        accumulator = warpReduction;
+      else
+        accumulator = vector::makeArithReduction(
+            rewriter, loc, reductionOp.kind(), accumulator, warpReduction);
+    }
+
+    rewriter.replaceOp(reductionOp, accumulator);
+    return success();
+  }
+};
+
```

Comment:
This is fine for now but I think eventually we should make this use the same pattern and interface as the rest of the unroll vector patterns:
https://github.com/llvm/llvm-project/blob/1768cb3a674ad657dd3eccd001f0a3e9a7675af6/mlir/lib/Dialect/Vector/Transforms/VectorUnrollDistribute.cpp#L210

This way it would be independent of Warp ops and can be used for other things. (and can be upstreamed independently.)
It is fine if you prefer to leave it like that for now to make progress on other stuff but I would at least add a comment.